airbyte-integrations/connectors/destination-bigquery/src/main/kotlin/io/airbyte/integrations/destination/bigquery/BigQueryBulkLoader.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.bigquery
+
+import com.google.cloud.bigquery.BigQuery
+import com.google.cloud.bigquery.FormatOptions
+import com.google.cloud.bigquery.JobInfo
+import com.google.cloud.bigquery.LoadJobConfiguration
+import io.airbyte.cdk.load.command.DestinationCatalog
+import io.airbyte.cdk.load.command.DestinationStream
+import io.airbyte.cdk.load.file.s3.S3KotlinClient
+import io.airbyte.cdk.load.file.s3.S3Object
+import io.airbyte.cdk.load.message.StreamKey
+import io.airbyte.cdk.load.write.db.BulkLoader
+import io.airbyte.cdk.load.write.db.BulkLoaderFactory
+import io.airbyte.integrations.destination.bigquery.formatter.BigQueryRecordFormatter
+import io.airbyte.integrations.destination.bigquery.spec.BigqueryConfiguration
+import io.airbyte.integrations.destination.bigquery.write.TempUtils
+import jakarta.inject.Singleton
+
+class BigQueryBulkLoader(
+    private val storageClient: S3KotlinClient,
+    private val bigQueryClient: BigQuery,
+    private val bigQueryConfiguration: BigqueryConfiguration,
+    private val stream: DestinationStream,
+) : BulkLoader<S3Object> {
```

Comment:
feels fine. We can always come back and make GcsObject wrap S3Object if we need to.